airbyte-server/src/main/java/io/airbyte/server/ConfigDumpImport.java

```diff
@@ -329,6 +337,36 @@ public void importDatabaseFromArchive(final Path storageRoot, final String airby
     }
   }
 
+  /**
+   * The deployment concept is specific to the environment that Airbyte is running in (not the data
+   * being imported). Thus, if there is a deployment in the imported data, we filter it out. In
+   * addition, before running the import, we look up the current deployment id, and make sure that
+   * that id is inserted when we run the import.
+   *
+   * @param postgresPersistence - database that we are importing into.
+   * @param metadataTableStream - stream of records to be imported into the metadata table.
+   * @return modified stream with old deployment id removed and correct deployment id inserted.
+   * @throws IOException - you never know when you IO.
+   */
+  private static Stream<JsonNode> replaceDeploymentMetadata(JobPersistence postgresPersistence,
+                                                            Stream<JsonNode> metadataTableStream)
+      throws IOException {
+    // filter out the deployment record from the import data, if it exists.
+    Stream<JsonNode> stream = metadataTableStream
+        .filter(record -> record.get(DefaultJobPersistence.METADATA_KEY_COL).asText().equals(DatabaseSchema.AIRBYTE_METADATA.toString()));
```

Comment:
@cgardens, this line looks problematic. The metadata records look like this:

```
Record: {"key":"server_uuid","value":"e895a584-7dbf-48ce-ace6-0bc9ea570c34"}
Record: {"key":"deployment_id","value":"55b923a9-42e5-4b0d-a5d2-b8d0316bfb2b"}
Record: {"key":"airbyte_version","value":"dev"}
Record: {"key":"2021-07-26T03:34:31.996221Z_init_db","value":"dev"}
```

The `METADATA_KEY_COL` column is never `AIRBYTE_METADATA`. So none of the metadata will be imported. Based on the comment, what you want is the following, right?

```suggestion
        .filter(record -> record.get(DefaultJobPersistence.METADATA_KEY_COL).asText().equals(DEPLOYMENT_ID_KEY));
```

I think this is going to cause a severe problem for anyone that tries to import their configs. Right now the database connection depends on the existence of the `service_uuid` metadata. If that record does not exist, the server and scheduler will be waiting for the connection forever.

Reply:
Fix here: #4977.